WcaOnRails/app/webpacker/components/Delegates/index.jsx

```diff
@@ -51,8 +52,23 @@ export default function Delegates() {
 
   const [adminMode, setAdminMode] = React.useState(false);
 
+  React.useEffect(() => {
```

Comment:
Why do we need this effect to enforce admin mode?

Reply:
Because this is an admin-only feature, and hence if a person with admin access opens `/delegates#all`, if the admin mode is off, it might look bad.

Reply:
Okay, makes sense from a design point of view. But then maybe `useEffect` is not the best way to achieve this, because it can uncontrollably overwrite some variable if you accidentally miss some side-effect.

What about the following instead?

```jsx
const [toggleAdmin, setToggleAdmin] = useState(false);

const adminMode = useMemo(() => toggleAdmin || (activeRegion === ALL_REGIONS && loggedInUserPermissions.canViewDelegateAdminPage), [toggleAdmin, activeRegion, loggedInUserPermissions]);
```

IMO that's much easier to read and easier to debug

Reply:
^ the `useMemo` isn't really needed.

Reply:
Okay, this looks better. I've implemented it. Also removed `useMemo` and renamed `adminMode` to `isAdminMode` as Kevin suggested.